lib/logstash/filters/aggregate.rb

```diff
@@ -384,10 +433,31 @@ def remove_expired_maps()
     min_timestamp = Time.now - @timeout
     
     @@mutex.synchronize do
+
       @@aggregate_maps.delete_if do |key, element| 
         if (element.creation_timestamp < min_timestamp)
-          if (@push_previous_map_as_event)
-            events_to_flush << LogStash::Event.new(element.map)
+          if (@push_previous_map_as_event) || (@push_map_as_event_on_timeout)
+            event = LogStash::Event.new(element.map)        
+            
+            if @timeout_task_id_field
+              event_to_yield[@timeout_task_id_field] = task_id
+            end
+
+            # Call code block if available
+            if @timeout_code
+              begin
+                @timeout_codeblock.call(event)
+                events_to_flush << event
+                noError = true
```

Comment:
As `noError` variable is never used in this method, I suggest you to remove this line.